.github/workflows/ci.yml

```diff
@@ -29,7 +29,16 @@ jobs:
       - run: npm run bootstrap -- --ci
       - run: npm run build
       - run: npm run test
-      - run: npm run e2e-test
+      - run: |
+          # Unfortunately GitHub Actions does not currently let us do something like
+          #     if: secrets.NPM_TOKEN != ''
+          # so simply skip the command if the env var is not set:
+          if [ -n $E2E_TEST_REFRESH_TOKEN ]; then
+          npm run e2e-test
+          fi
+        if:
+          # Testing for the secret is based on a bash expression, which isn't portable. 
+          ${{ matrix.os == 'ubuntu-20.04' }}
```

Comment:
This does mean that the end-to-end tests won't run at all on Windows any more, right? How much of a problem do you expect that to be? (I guess there's not much difference between the different OS's?)

Reply:
That's correct, I figured if the unit tests did run on all OS, the risks of the e2e tests failing only on Windows were relatively low, and at least low enough that it's not really worth working around that by writing a custom portable node script to replace the bash script :).